see_profile/__init__.py

```diff
@@ -0,0 +1,91 @@
+import cProfile
+import logging
+import pstats
+import sqlparse
+from io import StringIO
+
+from django.core.exceptions import MiddlewareNotUsed
+from django.db import connection
+
+from security.middleware import BaseMiddleware
+
+
+class ProfilingMiddleware(BaseMiddleware):
+    """
+    Adds the ability to profile requests via a header.
+
+    Usage:
+    Add the middleware to the MIDDLEWARE list. New boolean setting
+    "ENABLE_PROFILING" will be required to be set in the settings file. When
+    set to False, the middleware will deactivate itself. When set to True, the
+    middleware will be active.
+
+    When the middleware is active, it will log the data for any request that
+    supplies the X-Profile header in the HTTP request. This data will be logged
+    to the 'profiling' logger, so in order to see the results of this profiling
+    the Django logging will need to configure handlers for the 'profiling'
+    logger. Profiling will be configured at the DEBUG level.
+    """
+
+    REQUIRED_SETTINGS = ('ENABLE_PROFILING', 'DEBUG')
+    request_separator = f"\n{'=' * 80}\n"
+    query_separator = f"\n{'*' * 80}\n"
+
+    def __init__(self, get_response=None):
+        super().__init__(get_response)
+        if not self.enable_profiling:
+            raise MiddlewareNotUsed()
+        self.logger = logging.getLogger('profiling')
+
+    def load_setting(self, setting, value):
+        setattr(self, setting.lower(), value)
+
+    def format_queries(self, queries):
+        return self.query_separator.join(
+            sqlparse.format(query['sql'], reindent=True, keyword_case='upper')
+            for query in queries
+        )
+
+    def __call__(self, request):
+        # Only profile requests that have a 'X-Profile' HTTP header
+        if 'HTTP_X_PROFILE' not in request.META:
+            return self.get_response(request)
+
+        out = StringIO()
+        out.write(self.request_separator)
+
+        # Add method & path info to differentiate requests
+        out.write(
+            f"{request.method} {request.path}"
```

Comment:
Minor but do we want to add a newline after this? right now the log looks like
```
GET /api/v2/business-units/         319209 function calls (316528 primitive calls) in 0.810 seconds

   Ordered by: cumulative time
   List reduced from 1306 to 128 due to restriction <128>

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
        1    0.000    0.000    0.811    0.811 django/core/handlers/exception.py:38(inner)
        1    0.000    0.000    0.811    0.811 django/core/handlers/base.py:157(_get_response)
...
```
and it looks like the "319209 function calls" part is supposed to start on its own line?

Reply:
That's fair. Might be worth adding a newline.

Reply:
Done.